### PR TITLE
Support `concurrency` option in `OutputComparer`

### LINKED ISSUE #354: OutputComparer: add API to limit `max_pool` size
It can be convenient when comparing against engines with limitation (e.g., memory size)

### PR DESCRIPTION
**Pull request by asi1024**
Close #354.

~TODO: docstring~

### CODE REVIEW

File: pytorch_pfn_extras/utils/comparer.py [hunk 1]
```diff
@@ -93,6 +93,7 @@ class OutputsComparer:
     def __init__(
             self, engines, to_compare_keys=None, *,
             compare_fn=_default_comparer,
+            max_pool=None,
```
Comment (kmaehashi) [inline pytorch_pfn_extras/utils/comparer.py:96]: Maybe naming it `concurrency` is better?
```suggestion
            concurrency=None,
```

**Comment by asi1024** [comment]
PTAL.

File: pytorch_pfn_extras/utils/comparer.py [hunk 2]
```diff
@@ -105,6 +106,9 @@ def __init__(
                 A set of keys of output dict to compare.
             compare_fn (function):
                 Comparison function. Default is ``get_default_comparer()``.
+            concurrency (int, optional):
+                The upper bound limit on the number of workers that run concurrently.
```
Comment (emcastillo) [inline pytorch_pfn_extras/utils/comparer.py:110]:
```suggestion
                Maximum number of workers that run concurrently.
```

Maybe this is a bit clearer? just a nit.

Comment (asi1024) [reply pytorch_pfn_extras/utils/comparer.py]: Fixed. Thanks!

**Review by emcastillo — APPROVED** [review]

**Pull request merged**